OpenRA.Mods.Common/Activities/Move/MoveWithinRange.cs

```diff
@@ -53,5 +54,18 @@ bool AtCorrectRange(WPos origin)
 		{
 			return Target.IsInRange(origin, maxRange) && !Target.IsInRange(origin, minRange);
 		}
+
+		bool InCorrectLayer(Actor self)
```

Comment:
Somewhere, a blog post about how not to do separation of concerns is being written where we're every single reference. :)